concurrent/src/main/java/io/airlift/concurrent/Threads.java

```diff
@@ -84,18 +84,24 @@ public Thread newThread(Runnable runnable)
 
         public GroupedThreadFactory(String name)
         {
-            this(new ThreadGroup(name));
+            this.threadGroup =  new ThreadGroup(name);
         }
 
-        public GroupedThreadFactory(ThreadGroup threadGroup)
+        @Override
+        public Thread newThread(Runnable runnable)
         {
-            this.threadGroup = threadGroup;
+            return new Thread(threadGroup, runnable);
         }
 
         @Override
-        public Thread newThread(Runnable runnable)
+        protected void finalize()
         {
-            return new Thread(threadGroup, runnable);
+            try {
+                threadGroup.destroy();
```

Comment:
This can only be called after all threads in the group are dead. Does this guarantee that will happen?